### DISCUSSION TITLE
[Enhancement] [input-number] 允许 el-input-number 使用 el-input 里的部分Api

### DISCUSSION THREAD

**Discussion by skiyee** [element-plus/element-plus#17145] [Feature Request]
Feature type: **`Enhancement`**

## Details
* Enhanced component name: `el-input-number`

## Description
允许el-input-number使用el-input里部分API

Attributes:`formatter`、`parser` 等
Slots: `prepend`、`append` 等

就像 `ant` 中的 `input-number` 组件一样：https://www.antdv.com/components/input-number

**Comment by xxholly32** [comment]
感觉上可以~

**Comment by jinceon** [comment]
我就没搞懂，为什么要单独拆一个input-number出来

**Reply by warmthsea** [reply]
> 我就没搞懂，为什么要单独拆一个input-number出来

拆开比较好 input-number 存在大量代码 普通 input 并不需要，这样普通的 input 会更简洁更好维护

**Reply by jinceon** [reply]
但问题是，哪个input里有的功能，input-number不该有？

**Comment by jinceon** [comment]
![image](https://github.com/element-plus/element-plus/assets/1641265/066beac5-bb17-4a0b-a1b9-697a0d540ec7)
![image](https://github.com/element-plus/element-plus/assets/1641265/cdf98a89-6936-4d97-95fe-1d6d948eee9c)
![image](https://github.com/element-plus/element-plus/assets/1641265/c307dfa1-f189-4c11-979f-f3e62485eef3)
input-number不配有 prepend append 吗？
input-number的controls真的是个非常糟糕的设计。我理解的只有输入范围非常小，或者像slider那样拖拽到一个范围然后使用controls +-微调。不如的话，哪怕就输入一个1-10的数字，谁会用鼠标在那点10下？直接用键盘输入不香吗？

**Reply by warmthsea** [reply]
PR，Welcome.

**Reply by xxholly32** [reply]
嗯，当讨论一个问题的时候尽可能不要衍生另一个问题，如果有需要的话可以单开issues或discussion

> [关于 Yak Shaving](https://antfu.me/posts/about-yak-shaving-zh#yak-shaving)

**Reply by jinceon** [reply]
> 嗯，当讨论一个问题的时候尽可能不要衍生另一个问题，如果有需要的话可以单开issues或discussion
> 
> > [关于 Yak Shaving](https://antfu.me/posts/about-yak-shaving-zh#yak-shaving)

这就是同一个问题。
input有a,b功能，input有c,d功能。
现在提议在input-number也加上a,b,加来加去，不就是变成了一个大而全的组件
